Geometry/ForwardCommonData/plugins/dd4hep/DDTotemAngular.cc

```diff
@@ -0,0 +1,62 @@
+#include "DataFormats/Math/interface/GeantUnits.h"
+#include "DetectorDescription/Core/interface/DDSplit.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DD4hep/DetFactoryHelper.h"
+
+#define EDM_ML_DEBUG
+using namespace geant_units::operators;
+
+static long algorithm(dd4hep::Detector& /* description */,
+                      cms::DDParsingContext& ctxt,
+                      xml_h e,
+                      dd4hep::SensitiveDetector& /* sens */) {
+  cms::DDNamespace ns(ctxt, e, true);
+  cms::DDAlgoArguments args(ctxt, e);
+  double startAngle = args.value<double>("startAngle");
+  double stepAngle = args.value<double>("stepAngle");
+  double zoffset = args.value<double>("zoffset");
+  double roffset = args.value<double>("roffset");
+  int n = args.value<int>("n");
+  int startCopyNo = args.value<int>("startCopyNo");
+  int incrCopyNo = args.value<int>("incrCopyNo");
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("ForwardGeom") << "DDTotemAngular: Parameters for positioning-- " << n << " copies in steps of "
+                                  << convertRadToDeg(stepAngle) << " from " << convertRadToDeg(startAngle)
+                                  << " \tZoffset " << zoffset << " \tRoffset " << roffset
+                                  << "\tStart and inremental copy nos " << startCopyNo << ", " << incrCopyNo;
+#endif
+  std::string childName = args.value<std::string>("ChildName");
+  dd4hep::Volume parent = ns.volume(args.parentName());
+  dd4hep::Volume child = ns.volume(ns.prepend(childName));
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("ForwardGeom") << "DDTotemAngular debug: Parent " << parent.name() << "\tChild " << child.name();
+#endif
+
+  double phi = startAngle;
+  int copyNo = startCopyNo;
+
+  for (int ii = 0; ii < n; ii++) {
+    double phitmp = phi;
+    if (phitmp >= 2._pi)
+      phitmp -= 2._pi;
```

Comment:
We have a class for this purpose:
```
#include "DataFormats/GeometryVector/interface/Phi.h"
Geom::Phi0To2pi<double> phitmp = phi;
```
`phitmp` would automatically be kept in the range 0-2pi, and the case of <-2pi is taken care of, which line 41 misses.